src/mlpack/methods/adaboost/adaboost_fit_main.cpp

```diff
@@ -0,0 +1,180 @@
+/**
+ * @file methods/adaboost/adaboost_fit_main.cpp
+ * @author Udit Saxena
+ *
+ * Implementation of the AdaBoost main program.
+ *
+ * @code
+ * @article{Schapire:1999:IBA:337859.337870,
+ *   author = {Schapire, Robert E. and Singer, Yoram},
+ *   title = {Improved Boosting Algorithms Using Confidence-rated Predictions},
+ *   journal = {Machine Learning},
+ *   issue_date = {Dec. 1999},
+ *   volume = {37},
+ *   number = {3},
+ *   month = dec,
+ *   year = {1999},
+ *   issn = {0885-6125},
+ *   pages = {297--336},
+ *   numpages = {40},
+ *   url = {http://dx.doi.org/10.1023/A:1007614523901},
+ *   doi = {10.1023/A:1007614523901},
+ *   acmid = {337870},
+ *   publisher = {Kluwer Academic Publishers},
+ *   address = {Hingham, MA, USA},
+ *   keywords = {boosting algorithms, decision trees, multiclass classification,
+ *   output coding}
+ * @endcode
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include <mlpack/prereqs.hpp>
+#include <mlpack/core/util/io.hpp>
+
+#ifdef BINDING_NAME
+  #undef BINDING_NAME
+#endif
+#define BINDING_NAME adaboost_fit
+
+#include <mlpack/core/util/mlpack_main.hpp>
+#include <mlpack/core/data/normalize_labels.hpp>
+#include "adaboost.hpp"
+#include "adaboost_model.hpp"
+
+using namespace mlpack;
+using namespace std;
+using namespace arma;
+using namespace mlpack::adaboost;
+using namespace mlpack::tree;
+using namespace mlpack::perceptron;
+using namespace mlpack::util;
+
+// Program Name.
+BINDING_USER_NAME("AdaBoost Training");
+
+// Short description.
+BINDING_SHORT_DESC(
+    "Training AdaBoost model.");
+
+// Long description.
+BINDING_LONG_DESC("");
```

Comment:
You had mentioned that we can remove this macro eventually entirely---maybe before we merge this PR we should be sure to write down that intention somewhere?  Perhaps in a separate issue or something?

Reply:
Actually, I think we can use that for now. We can use short description as the one-liners we are planning to print and the long description as the description that is printed just above the example. Take a look at `adaboost_train_main.cpp`

Reply:
Right, that makes sense, but how do we know in the binding code which binding to use the `LONG_DESCRIPTION()` from?  I guess you are just looking for the name `train`?  (I guess that's ok for now, but we should probably document that in the `LONG_DESCRIPTION()` documentation.)

Reply:
actually, whatever method is passed first in the group_bindings() macro is expected to have a long description, but I agree, this should be documented. done :+1:

Reply:
Yep, that's just fine.  I think eventually we will end up cleaning things up a bit once all the bindings are grouped, but using the long description from the first binding is just fine for now. :+1: